### DISCUSSION TITLE
crosshole ERT

### DISCUSSION THREAD

**Discussion by anzhijiang** [gimli-org/pyGIMLi#338] [Ideas]
### Problem description
The  example "plot_01_ert_2d_mod_inv.py" runs well. Now I want to creat crosshole 2D ERT modeling and inversion. So how to create the ERT scheme according to this example code?

### Your environment

**Operating system**:  Linux
**Python version**: ,3.6
**pyGIMLi version**: 1.2.1
**Way of installation**: Conda package

**Comment by halbmy** [comment]
Unlike for surface ERT, there are no standard arrays that can be created by `ert.createData()`. However, it would be interesting to collect "typical crosshole ERT arrays" and generators, at least for two boreholes. So anyone who can contribute is invited. One could start looking at open-source papers published along with open data.

**Comment by bullcat123** [comment]
following @anzhijiang question, is it possible to import a crossholes array (as a dat or csv file) to use for forward modeling in pygimli?

**Reply by halbmy** [reply]
Of course you can, which kind of format do you have?

You can also create it by some for loops just like in the crosshole traveltime example, using a/b/m/n instead of s/g.

**Reply by bullcat123** [reply]
I have a csv file that contain 'a,b,m,n' headers, and in each line the serials numbers of the electrodes (i have 16 electrode in total, num 1-16):

a, b, m, n 
1, 2, 5, 6
...

and than should i still use the `ert.createData()` command?

**Reply by halbmy** [reply]
I guess it is the easiest to create a file and read it by `pg.DataContainerERT(filename)`:
```
16 # electrodes
#x z
0 -1
0 -1.5 
...
111 # number of data
# a b m n
1 2 5 6
...
```

**Comment by halbmy** [comment]
I got an email on this issue about problems plotting the pseudosection but I can't see the text here. 

There is no unique procedure for plotting as for surface ERT data as there are many different types of arrays. I suggest using codes like
```
pg.viewer.mpl.showDataContainerAsMatrix(data, "a", "m", "rhoa")
```
